Split assignment and unassignment tasks

pezholio opened the pull request:
This splits the assignment and unassignment of users to the 1st line support group into two distinct tasks. When a user comes into the support rotation, they are added to the 1st line group in the morning, and when a user is coming off the support rotation, they are taken out of the group in the afternoon. This gives the user who is coming off support time to hand over tickets they are assigned to, without them being automatically unassigned when they are moved out of the group.

pezholio committed:
Set first line dev support group in helper

This guards agains the .env file value being set to anything else and causing
confusion when the tests don't pass.

pezholio commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Stop unassigning tickets on Zendesk as soon as support rotas change](https://trello.com/c/Dn4Gdyzk/90-stop-unassigning-tickets-on-zendesk-as-soon-as-support-rotas-change)

erbridge commented on lib/huntaway.rb:
```diff
@@ -9,35 +9,26 @@ class Huntaway
+    group_memberships.each do |group_membership|
       if group_membership["group_id"] == FIRST_LINE_DEV_SUPPORT_GROUP_ID
-        @existing_group_memberships << group_membership["user_id"]
-        unless @current_support_user_ids.include?(group_membership["user_id"])
+        unless current_support_user_ids.include?(group_membership["user_id"])
           group_membership.destroy!
         end
```
This we could reduce the nesting by flipping the conditional to a guard clause?

erbridge commented on .github/workflows/unassign.yml:
```diff
@@ -0,0 +1,29 @@
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
```
```suggestion
    - uses: actions/checkout@v2
```

Prefer v2 unless you're doing any Git operations.

erbridge commented on .github/workflows/unassign.yml:
```diff
@@ -0,0 +1,29 @@
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Ruby 2.6
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
```
Can we use the action that picks up `.ruby_version`?

erbridge commented on .github/workflows/unassign.yml:
```diff
@@ -0,0 +1,29 @@
+      uses: actions/setup-ruby@v1
+      with:
+        ruby-version: 2.6.x
+    - name: Unassign outgoing support user from the first line support group
+      run: |
+        gem install bundler
+        bundle install --jobs 4 --retry 3
+        bundle exec rake huntaway:unassign_extra_users_from_group
```
Would be nice to split the dependency installation into a different step, to make viewing the logs easier.

erbridge commented on .github/workflows/unassign.yml:
```diff
@@ -0,0 +1,29 @@
+
+on:
+  schedule:
+    - cron: '0 14 * * 1-5'
```
This is likely to be confusing when we shift timezones (right now, we're in BST, so +1, i.e., this actually runs at 13:00).

erbridge replied:
You can also use `MON-FRI` for more clarity on what the last numbers mean.

pezholio replied:
Yeah, I've pointed out that these run at UTC. I've come across this before and I'm pretty sure you can't specify what timezone to use in actions.

erbridge replied:
You can't. I think erring for later, rather than earlier is the thing to do for this one, so setting to 15:00 UTC.

pezholio replied:
Yeah, good plan

erbridge reviewed (commented):
Some small tweaks.

pezholio committed:
Split assignment and unassignment into two methods

This allows us to assign and unassign at different times, allowing time for
tickets to be handed over from the outgoing developer and incoming
developer

pezholio committed:
Improve Github actions

erbridge commented on .github/workflows/assign.yml:
```diff
@@ -0,0 +1,34 @@
+      run: |
+        gem install bundler
+        bundle install --jobs 4 --retry 3 --path ./vendor/bundle
+    - name: Assign current support user to correct group
```
```suggestion
    - name: Assign incoming support user to the first line support group
```

pezholio replied:
👍

pezholio committed:
Add task to unassign the outgoing support user

This runs at 2pm (UTC) every day and unassigns all users from the 1st line
support group if they are no longer in the support rotation.

erbridge reviewed (approved):
Minor suggestion on one of the step names. Not a blocker. Otherwise good to go.

pezholio committed:
Update crontab to be clearer about when it runs

# Conflicts:
#	.github/workflows/unassign.yml

pezholio committed:
Update Rakefile to have two distinct tasks

One task now handles the group assignment, and the other handles the
group unassignment

pezholio committed:
Update docs
